apps/server/src/modules/h5p-editor/contentStorage/contentStorage.ts

```diff
@@ -1,173 +1,156 @@
 /* eslint-disable @typescript-eslint/no-unused-vars */
-import { Injectable } from '@nestjs/common';
-import { Readable, Stream } from 'stream';
-import * as fs from 'node:fs';
-import * as fsPromises from 'node:fs/promises';
-import { getAllFiles } from 'get-all-files';
 import {
 	ContentId,
 	IContentMetadata,
 	IContentStorage,
-	LibraryName,
 	IFileStats,
 	ILibraryName,
 	IUser,
+	LibraryName,
 	Permission,
 	streamToString,
 } from '@lumieducation/h5p-server';
-import path from 'path';
-import rimraf from 'rimraf';
+import { Injectable, InternalServerErrorException, NotFoundException } from '@nestjs/common';
+import { FileDto } from '@src/modules/files-storage/dto';
+import { Readable } from 'stream';
+import { S3ClientAdapter } from '../../files-storage/client/s3-client.adapter';
+import { H5PContent } from './h5p-content.entity';
+import { H5PContentRepo } from './h5p-content.repo';
 
 @Injectable()
 export class ContentStorage implements IContentStorage {
-	constructor(
-		protected contentPath: string,
-		protected options?: {
-			invalidCharactersRegexp?: RegExp;
-			maxPathLength?: number;
-		}
-	) {}
+	constructor(private readonly repo: H5PContentRepo, private readonly storageClient: S3ClientAdapter) {}
 
 	public async addContent(
 		metadata: IContentMetadata,
-		content: unknown,
+		content: string | undefined,
 		user: IUser,
 		contentId?: ContentId | undefined
 	): Promise<ContentId> {
-		if (contentId === null || contentId === undefined) {
-			contentId = await this.createContentId();
+		if (!metadata.defaultLanguage) {
+			metadata.defaultLanguage = metadata.language;
 		}
+
 		try {
-			fs.existsSync(path.join(this.getContentPath(), contentId.toString()));
-			await this.existsOrCreateDir(contentId);
-			await fsPromises.writeFile(
-				path.join(this.getContentPath(), contentId.toString(), 'h5p.json'),
-				JSON.stringify(metadata)
-			);
-			await fsPromises.writeFile(
-				path.join(this.getContentPath(), contentId.toString(), 'content.json'),
-				JSON.stringify(content)
-			);
-		} catch (error) {
-			if (fs.existsSync(path.join(this.getContentPath(), contentId.toString()))) {
-				rimraf.sync(path.join(this.getContentPath(), contentId.toString()));
+			let h5pContent: H5PContent;
+
+			if (contentId) {
+				h5pContent = await this.repo.findById(contentId);
+				h5pContent.metadata = metadata;
+				h5pContent.content = content;
+				await this.repo.save(h5pContent);
+			} else {
+				h5pContent = new H5PContent({ metadata, content });
+				await this.repo.createContent(h5pContent);
 			}
 
-			// eslint-disable-next-line @typescript-eslint/restrict-template-expressions, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
-			throw new Error(`Error creating content.${error.toString()}`);
+			return h5pContent.id;
+		} catch (error) {
+			throw new InternalServerErrorException(error, 'ContentStorage:addContent');
```

Comment:
Is there a rule which function gets this try-catch-block and throws an InternalServerError? Because e.g. addContent has it, addFile does not have it. 

IMO it is not necessary to have it just to add this string ('ContentStorage:addContent') to the error. Seems like a lot of overhead just to do that, but I'm not sure if there was some decision that we want that....

Reply:
This was inspired by the S3ClientAdapter, where errors that could get thrown from multiple sources are consolidated in the same way.